### PR TITLE
Add support for sparse checkouts

### PR DESCRIPTION
**Pull request by jonnelaaglandwinder**
I've taken a crack at implementing a sparse checkout option suggested in #172.
It may prove useful for large monorepos, especially if a --filter option for fetch were to be implemented as well.

I've put a temporary escape hatch in the test script, since it would fail without the test data branch being present, but it should pass with those lines removed when the test data exists. ([test-data/v2/sparse-checkout](https://github.com/Nocturne92/checkout/tree/test-data/v2%2Fsparse-checkout))

### CODE REVIEW

File: __test__/verify-sparse-checkout.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+## REMOVE THIS
+exit 0
+## REMOVE THIS
```
Comment (rcowsill) [inline __test__/verify-sparse-checkout.sh:5]: Should this be removed now?

**Comment by arcanis** [comment]
Bump @thboop? This would be extremely useful for repositories that check-in (somewhat) large artifacts, like their zipped dependencies, which is a somewhat common pattern with Yarn.

**Comment by Wrathchyld** [comment]
action.yml

**Comment by tyson-benson** [comment]
This would be a great addition 👍 
It would be a valuable capability for those that want to adopt mono-repo design. I hope this PR gets the attention it deserves.

**Comment by arcanis** [comment]
Ping @TingluoHuang? Are there any blockers?

File: __test__/verify-sparse-checkout.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (austinpray-mixpanel) [inline __test__/verify-sparse-checkout.sh:6]:
```suggestion
```

**Comment by blu3mania** [comment]
Shouldn't lfs.fetchinclude also be set or use "-I" (or "--include") parameter to make sure LFS only fetches from included paths?

**Comment by guitarrapc** [comment]
Is any blocker exists? I'm eager to use this feature but wait for half a year.

**Comment by derekperkins** [comment]
Any feedback from @actions-runtime?

**Comment by hyoretsu** [comment]
This would be great for monorepos y'know.

**Comment by zchenyu** [comment]
Another friendly bump on this. +1 for this feature

**Comment by dfdez** [comment]
Hey everyone! I see that this PR has been frozen for a while...

In my company we need to use the `sparse-checkout` functionality, that is why I have invested time in order to retake this. 

Here you can find a [PR](https://github.com/actions/checkout/pull/1317) with an updated version to add support for the `sparse-checkout`!

I have to mention that this PR has helped me in order to make mine so thanks @jonnelaaglandwinder to start this, let's hope we can finally have this functionality implemented!

**Review by Spiritaine — APPROVED** [review]

**Comment by dfdez** [comment]
This PR can be closed due https://github.com/actions/checkout/pull/1369

**Review by Spiritaine — APPROVED** [review]

**Pull request closed without merging**